# **1. Chargement et téléchargement des libraries**

Installation du package TCGAbiolinks : à faire une fois

In [1]:
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

BiocManager::install(c(
  "TCGAbiolinks",
  "SummarizedExperiment"
))

install.packages(c("dplyr", "ggplot2", "pheatmap"))

'getOption("repos")' replaces Bioconductor standard repositories, see
'help("repositories", package = "BiocManager")' for details.
Replacement repositories:
    CRAN: https://cran.r-project.org

Bioconductor version 3.21 (BiocManager 1.30.27), R 4.5.2 (2025-10-31 ucrt)

Warning message:
"package(s) not installed when version(s) same as or greater than current; use
  `force = TRUE` to re-install: 'SummarizedExperiment'"
Installing package(s) 'TCGAbiolinks'

installation des dépendances 'dbplyr', 'filelock', 'httr2', 'BiocFileCache', 'selectr', 'clipr', 'vroom', 'tzdb', 'downloader', 'biomaRt', 'rvest', 'TCGAbiolinksGUI.data', 'readr'





  Une version binaire est disponible mais la version du source est plus
  récente:
              binary source needs_compilation
BiocFileCache 2.16.1 2.16.2             FALSE

le package 'dbplyr' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'filelock' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'httr2' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'selectr' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'clipr' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'vroom' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'tzdb' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'downloader' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'biomaRt' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'rvest' a été décompressé et les so

installation des packages sources 'BiocFileCache', 'TCGAbiolinksGUI.data'


Old packages: 'BH', 'digest', 'e1071', 'emmeans', 'future.apply', 'ggsci',
  'htmltools', 'isoband', 'knitr', 'leidenbase', 'parallelly', 'proxy',
  'RcppArmadillo', 'reformulas', 'Rsamtools', 'Seurat', 'SeuratObject',
  'shiny', 'tidyr', 'xfun', 'yaml', 'yulab.utils', 'zoo'

Installation des packages dans 'C:/Users/assou/AppData/Local/R/win-library/4.5'
(car 'lib' n'est pas spécifié)



le package 'dplyr' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'ggplot2' a été décompressé et les sommes MD5 ont été vérifiées avec succés
le package 'pheatmap' a été décompressé et les sommes MD5 ont été vérifiées avec succés

Les packages binaires téléchargés sont dans
	C:\Users\assou\AppData\Local\Temp\RtmpqMwgWJ\downloaded_packages


chargement des libraries

In [ ]:
library(TCGAbiolinks)
library(SummarizedExperiment)
library(dplyr)
library(ggplot2)
library(pheatmap)

set.seed(555)

# **2. Chargement du dataset TCGA thyroide**

### _**téléchargement des données TCGA**_
- projet THCA : cancer de la thyroide 
- seulement les données de quantification d'expression génique et de transcriptome profiling 

In [6]:
query <- GDCquery(
  project = "TCGA-THCA",
  data.category = "Transcriptome Profiling",
  data.type = "Gene Expression Quantification",
  workflow.type = "STAR - Counts"
)

GDCdownload(query)
se <- GDCprepare(query)

--------------------------------------

o GDCquery: Searching in GDC database

--------------------------------------

Genome of reference: hg38

--------------------------------------------

oo Accessing GDC. This might take a while...

--------------------------------------------

ooo Project: TCGA-THCA

--------------------

oo Filtering results

--------------------

ooo By data.type

ooo By workflow.type

----------------

oo Checking data

----------------

ooo Checking if there are duplicated cases

ooo Checking if there are results for the query

-------------------

o Preparing output

-------------------


Of the 572 files for download 572 already exist.

All samples have been already downloaded



|====================================================|100%                      Completed after 1 m 


Starting to add information to samples

 => Add clinical information to samples

 => Adding TCGA molecular information from marker papers

 => Information will have prefix 'paper_' 

thca subtype information from:doi:10.1016/j.cell.2014.09.050

Available assays in SummarizedExperiment : 
  => unstranded
  => stranded_first
  => stranded_second
  => tpm_unstrand
  => fpkm_unstrand
  => fpkm_uq_unstrand



In [ ]:
se

class: RangedSummarizedExperiment 
dim: 60660 572 
metadata(1): data_release
assays(6): unstranded stranded_first ... fpkm_unstrand fpkm_uq_unstrand
rownames(60660): ENSG00000000003.15 ENSG00000000005.6 ...
  ENSG00000288674.1 ENSG00000288675.1
rowData names(10): source type ... hgnc_id havana_gene
colnames(572): TCGA-DJ-A2Q6-01A-11R-A18C-07
  TCGA-FK-A3SE-01A-11R-A22L-07 ... TCGA-DJ-A2PX-01A-11R-A18C-07
  TCGA-EL-A3ZS-01A-12R-A23N-07
colData names(241): barcode patient ... paper_TERT_AC paper_TERT_Q

### _**load des données cliniques**_

In [56]:
clinique <- GDCquery_clinic(
  project = "TCGA-THCA",
  type = "clinical"
)

head(clinique)
names(clinique)

#lister tous les éléments de ajcc_pathologic_stage
table(clinique$ajcc_pathologic_stage)

project   submitter_id synchronous_malignancy ajcc_pathologic_stage
1 TCGA-THCA TCGA-EM-A2OZ No                     Stage I              
2 TCGA-THCA TCGA-EM-A22J No                     Stage II             
3 TCGA-THCA TCGA-DJ-A2PR No                     Stage I              
4 TCGA-THCA TCGA-ET-A40S No                     Stage II             
5 TCGA-THCA TCGA-ET-A39M No                     Stage I              
6 TCGA-THCA TCGA-ET-A40Q No                     Stage I              
  days_to_diagnosis laterality created_datetime last_known_disease_status
1 0                 Midline    NA               NA                       
2 0                 Left       NA               NA                       
3 0                 Bilateral  NA               NA                       
4 0                 Left       NA               NA                       
5 0                 Left       NA               NA                       
6 0                 Right      NA               NA                       
  tissue_or_organ_of_origin age_at_diagnosis ⋯
1 Thyroid gland             24195            ⋯
2 Thyroid gland             20539            ⋯
3 Thyroid gland              9914            ⋯
4 Thyroid gland             22733            ⋯
5 Thyroid gland             12441            ⋯
6 Thyroid gland             14136            ⋯
  treatments_radiation_treatment_anatomic_sites
1 NULL                                         
2 NULL;Primary Tumor Field                     
3 NULL                                         
4 NULL                                         
5 NULL;Primary Tumor Field                     
6 NULL;Primary Tumor Field                     
  treatments_radiation_treatment_outcome
1 NA                                    
2 NA;Complete Response                  
3 NA                                    
4 Not Reported;NA                       
5 NA;Complete Response                  
6 NA;Unknown                            
  treatments_radiation_treatment_dose_units
1 NA                                       
2 NA;mCi                                   
3 NA                                       
4 NA                                       
5 NA                                       
6 NA;mCi                                   
  treatments_radiation_number_of_fractions
1 NA                                      
2 NA                                      
3 NA                                      
4 NA                                      
5 NA;1                                    
6 NA;1                                    
  treatments_radiation_radiosensitizing_agent
1 NA                                         
2 NA                                         
3 NA                                         
4 NA                                         
5 NA                                         
6 NA                                         
  treatments_radiation_course_number
1 NA                                
2 NA                                
3 NA                                
4 NA                                
5 NA                                
6 NA                                
  treatments_radiation_clinical_trial_indicator
1 NA                                           
2 NA                                           
3 NA                                           
4 NA                                           
5 NA                                           
6 NA                                           
  treatments_radiation_route_of_administration
1 NULL                                        
2 NULL                                        
3 NULL                                        
4 NULL                                        
5 NULL                                        
6 NULL                                        
  treatments_radiation_initial_disease_status bcr_patient_barcode
1 NA                                          TCGA-EM-A2OZ       
2 NA                                          TCGA-EM-A

[1] "project"                                           
 [2] "submitter_id"                                      
 [3] "synchronous_malignancy"                            
 [4] "ajcc_pathologic_stage"                             
 [5] "days_to_diagnosis"                                 
 [6] "laterality"                                        
 [7] "created_datetime"                                  
 [8] "last_known_disease_status"                         
 [9] "tissue_or_organ_of_origin"                         
[10] "age_at_diagnosis"                                  
[11] "primary_diagnosis"                                 
[12] "updated_datetime"                                  
[13] "prior_malignancy"                                  
[14] "year_of_diagnosis"                                 
[15] "state"                                             
[16] "prior_treatment"                                   
[17] "diagnosis_is_primary_disease"                      
[18] "days_to_last_known_disease_status"                 
[19] "ajcc_staging_system_edition"                       
[20] "ajcc_pathologic_t"                                 
[21] "days_to_recurrence"                                
[22] "morphology"                                        
[23] "ajcc_pathologic_n"                                 
[24] "ajcc_pathologic_m"                                 
[25] "residual_disease"                                  
[26] "classification_of_tumor"                           
[27] "tumor_focality"                                    
[28] "diagnosis_id"                                      
[29] "icd_10_code"                                       
[30] "site_of_resection_or_biopsy"                       
[31] "tumor_grade"                                       
[32] "progression_or_recurrence"                         
[33] "tumor_of_origin"                                   
[34] "cigarettes_per_day"                                
[35] "alcohol_history"                                   
[36] "exposure_id"                                       
[37] "exposure_type"                                     
[38] "alcohol_intensity"                                 
[39] "race"                                              
[40] "gender"                                            
[41] "ethnicity"                                         
[42] "vital_status"                                      
[43] "age_at_index"                                      
[44] "days_to_birth"                                     
[45] "year_of_birth"                                     
[46] "demographic_id"                                    
[47] "age_is_obfuscated"                                 
[48] "year_of_death"                                     
[49] "country_of_residence_at_enrollment"                
[50] "days_to_death"                                     
[51] "days_to_last_follow_up"                            
[52] "follow_ups_disease_response"                       
[53] "treatments_pharmaceutical_treatment_intent_type"   
[54] "treatments_pharmaceutical_treatment_id"            
[55] "treatments_pharmaceutical_treatment_type"          
[56] "treatments_pharmaceutical_treatment_or_therapy"    
[57] "treatments_pharmaceutical_therapeutic_agents"      
[58] "treatments_pharmaceutical_timepoint_category"      
[59] "treatments_pharmaceutical_prescribed_dose_units"   
[60] "treatments_pharmaceutical_pretreatment"            
[61] "treatments_pharmaceutical_prescribed_dose"         
[62] "treatments_pharmaceutical_days_to_treatment_end"   
[63] "treatments_pharmaceutical_treatment_dose"          
[64] "treatments_pharmaceutical_days_to_treatment_start" 
[65] "treatments_pharmaceutical_treatment_anatomic_sites"
[66] "treatments_pharmaceutical_treatment_outcome"       
[67] "treatments_pharmaceutical_treatment_dose_units"    
[68] "treatments_pharmaceutical_number_of_fractions"     
[69] "treatments_pharmaceutical_radiosensitizing_agent"  


  Stage I  Stage II Stage III  Stage IV Stage IVA Stage IVC 
      285        52       113         2        47         6 

# **3. exploration du dataset**

In [8]:
# Dimensions
se
dim(se)

class: RangedSummarizedExperiment 
dim: 60660 572 
metadata(1): data_release
assays(6): unstranded stranded_first ... fpkm_unstrand fpkm_uq_unstrand
rownames(60660): ENSG00000000003.15 ENSG00000000005.6 ...
  ENSG00000288674.1 ENSG00000288675.1
rowData names(10): source type ... hgnc_id havana_gene
colnames(572): TCGA-DJ-A2Q6-01A-11R-A18C-07
  TCGA-FK-A3SE-01A-11R-A22L-07 ... TCGA-DJ-A2PX-01A-11R-A18C-07
  TCGA-EL-A3ZS-01A-12R-A23N-07
colData names(241): barcode patient ... paper_TERT_AC paper_TERT_Q

[1] 60660   572

In [9]:
# Métadonnées
meta <- as.data.frame(colData(se))
head(meta)

barcode                      patient     
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA
TCGA-FY-A2QD-01A-11R-A19O-07 TCGA-FY-A2QD-01A-11R-A19O-07 TCGA-FY-A2QD
TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3GR
TCGA-EL-A3T7-01A-11R-A22L-07 TCGA-EL-A3T7-01A-11R-A22L-07 TCGA-EL-A3T7
                             sample           shortLetterCode
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6-01A TP             
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE-01A TP             
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA-01A TP             
TCGA-FY-A2QD-01A-11R-A19O-07 TCGA-FY-A2QD-01A TP             
TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3GR-01A TP             
TCGA-EL-A3T7-01A-11R-A22L-07 TCGA-EL-A3T7-01A TP             
                             definition          sample_submitter_id
TCGA-DJ-A2Q6-01A-11R-A18C-07 Primary solid Tumor TCGA-DJ-A2Q6-01A   
TCGA-FK-A3SE-01A-11R-A22L-07 Primary solid Tumor TCGA-FK-A3SE-01A   
TCGA-DJ-A2QA-01A-11R-A18C-07 Primary solid Tumor TCGA-DJ-A2QA-01A   
TCGA-FY-A2QD-01A-11R-A19O-07 Primary solid Tumor TCGA-FY-A2QD-01A   
TCGA-EL-A3GR-01A-21R-A206-07 Primary solid Tumor TCGA-EL-A3GR-01A   
TCGA-EL-A3T7-01A-11R-A22L-07 Primary solid Tumor TCGA-EL-A3T7-01A   
                             tumor_descriptor
TCGA-DJ-A2Q6-01A-11R-A18C-07 Primary         
TCGA-FK-A3SE-01A-11R-A22L-07 Primary         
TCGA-DJ-A2QA-01A-11R-A18C-07 Primary         
TCGA-FY-A2QD-01A-11R-A19O-07 Primary         
TCGA-EL-A3GR-01A-21R-A206-07 Primary         
TCGA-EL-A3T7-01A-11R-A22L-07 Primary         
                             sample_id                           
TCGA-DJ-A2Q6-01A-11R-A18C-07 74b9c0c8-ef36-4194-ab8c-d81d61a39705
TCGA-FK-A3SE-01A-11R-A22L-07 d3cc158d-413d-49e6-8d23-3ed89fdea94a
TCGA-DJ-A2QA-01A-11R-A18C-07 2277f3f2-411e-47ab-a2b9-f80fd33540a1
TCGA-FY-A2QD-01A-11R-A19O-07 08d3cb3b-9a80-4034-b06d-b37dd38c8b9b
TCGA-EL-A3GR-01A-21R-A206-07 12693f53-fa09-494c-82eb-bf735ea84e98
TCGA-EL-A3T7-01A-11R-A22L-07 76e3d9f4-513f-4a44-8918-4c29ce5fb305
                             pathology_report_uuid                submitter_id
TCGA-DJ-A2Q6-01A-11R-A18C-07 CCF825BD-7410-4ED2-B0E2-8B17BB5C5BBD TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07 3489196F-F9FD-4797-B4F2-0D66D420102A TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07 E3CC8058-B202-4433-8597-6BD196D04AAC TCGA-DJ-A2QA
TCGA-FY-A2QD-01A-11R-A19O-07 D4447081-2054-46B6-B83F-C0D2DF8C4AA1 TCGA-FY-A2QD
TCGA-EL-A3GR-01A-21R-A206-07 F2945CC2-AEE4-45F7-A640-4174EB0D37FD TCGA-EL-A3GR
TCGA-EL-A3T7-01A-11R-A22L-07 6B1080F8-AF82-4FE3-B91D-3CA2DBA4A37E TCGA-EL-A3T7
                             ⋯ paper_mutDriver_DCCF_CI95H
TCGA-DJ-A2Q6-01A-11R-A18C-07 ⋯ 1,000                     
TCGA-FK-A3SE-01A-11R-A22L-07 ⋯                           
TCGA-DJ-A2QA-01A-11R-A18C-07 ⋯ 1,000                     
TCGA-FY-A2QD-01A-11R-A19O-07 ⋯                           
TCGA-EL-A3GR-01A-21R-A206-07 ⋯ 1,000                     
TCGA-EL-A3T7-01A-11R-A22L-07 ⋯ 0,982                     
                             paper_mutDriver_CCF_gene paper_EXCLUDE_SAMPLE
TCGA-DJ-A2Q6-01A-11R-A18C-07 BRAF:1.000               0                   
TCGA-FK-A3SE-01A-11R-A22L-07                          0                   
TCGA-DJ-A2QA-01A-11R-A18C-07 BRAF:1.000               0                   
TCGA-FY-A2QD-01A-11R-A19O-07                          0                   
TCGA-EL-A3GR-01A-21R-A206-07 BRAF:0.910               0                   
TCGA-EL-A3T7-01A-11R-A22L-07 BRAF:0.850               0                   
                             paper_patient paper_TERT_DEPTH paper_TERT_POWER
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6  81005            1               
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE  44535            1               
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA  75617            1               
TCGA-FY-A2QD-01A-11R-A19O-07 TCGA-FY-A2

In [10]:
# Types d’échantillons
table(meta$sample_type)
table(meta$shortLetterCode)


         Metastatic       Primary Tumor Solid Tissue Normal 
                  8                 505                  59 


 NT  TM  TP 
 59   8 505 

# **4. Création de la matrice d'expression**

## *Loading de la matrice d'expresison brute*

In [11]:
expr <- assay(se, "unstranded")   # counts
head(expr)
dim(expr)

TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-FK-A3SE-01A-11R-A22L-07
ENSG00000000003.15 6156                         7191                        
ENSG00000000005.6     7                            3                        
ENSG00000000419.13 2540                         3815                        
ENSG00000000457.14 1452                          767                        
ENSG00000000460.17  297                          276                        
ENSG00000000938.13 1075                         1069                        
                   TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-FY-A2QD-01A-11R-A19O-07
ENSG00000000003.15 2219                         5185                        
ENSG00000000005.6     1                            0                        
ENSG00000000419.13  735                         2594                        
ENSG00000000457.14  305                          970                        
ENSG00000000460.17   28                          135                        
ENSG00000000938.13  303                           71                        
                   TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3T7-01A-11R-A22L-07
ENSG00000000003.15 4466                         9702                        
ENSG00000000005.6     3                            0                        
ENSG00000000419.13 1496                         2506                        
ENSG00000000457.14  820                         1036                        
ENSG00000000460.17  208                          315                        
ENSG00000000938.13  610                         1576                        
                   TCGA-EL-A3T7-11A-21R-A22L-07 TCGA-FE-A230-01A-11R-A14Y-07
ENSG00000000003.15 4546                         2449                        
ENSG00000000005.6    10                            0                        
ENSG00000000419.13 1854                          811                        
ENSG00000000457.14 1279                          619                        
ENSG00000000460.17  183                          194                        
ENSG00000000938.13  459                          446                        
                   TCGA-EM-A3FO-01A-11R-A21D-07 TCGA-DJ-A3VB-01A-11R-A23N-07 ⋯
ENSG00000000003.15 6336                         5997                         ⋯
ENSG00000000005.6     0                            2                         ⋯
ENSG00000000419.13 1892                         3074                         ⋯
ENSG00000000457.14  725                          856                         ⋯
ENSG00000000460.17  149                          317                         ⋯
ENSG00000000938.13  315                         1829                         ⋯
                   TCGA-EM-A2P1-01A-11R-A206-07 TCGA-EL-A3T1-01A-11R-A22L-07
ENSG00000000003.15 2691                         4533                        
ENSG00000000005.6     0                            0                        
ENSG00000000419.13 2227                         2406                        
ENSG00000000457.14  759                         1120                        
ENSG00000000460.17  279                          381                        
ENSG00000000938.13 1147                         1127                        
                   TCGA-EL-A3T1-11A-11R-A22L-07 TCGA-DJ-A2PS-01A-11R-A18C-07
ENSG00000000003.15 2931                         4471                        
ENSG00000000005.6     5                            0                        
ENSG00000000419.13 2209                         1832                        
ENSG00000000457.14  607                          922                        
ENSG00000000460.17  169                          226                        
ENSG00000000938.13 1169                          440                        
                   TCGA-EL-A3TB-11A-11R-A22L-07 TCGA-EL-A3TB-01A-11R-A22L-07
ENSG00000000003.15 5029                         4742                        
ENSG00000000005.6     4                            3                        
E

[1] 60660   572

_commentaire_ : la matrice d'expression de base a 572 échantillons : 
- **8** "Metastatic"
- **59** "Solid Tissue Normal"
- **505** "Primary Tumor"

Mais les __*Solid Tissue Normal*__ ne sont pas des contrôles sains, juste des échantillons de tissus normaux prélevés à côté de la tumeur, ils n'ont pas de stade tumoral. Il faut qu'on les enlève pour ne garder que les échantillons "Primary Tumor".

Pour les __*Metastasique*__, ce ne sont pas des tissus thyroidiens, et ont un profil d'expression reprogrammé et souvent issus de stades avancés. Du coup le modèle va apprendre seulement le site métastasique plutôt que la biologie du stade tumoral. En fait, on fait ici du data leakage parce que les métastases impliquent forcément un stade avancé.

Alors, on ne garde que les __*Primary Tumor*__ pour faire l'analyse et pouvoir prédire les stades tumoraux. C'est ce que fait la cellule de code suivante :

In [13]:
se_tumor <- se[, meta$sample_type == "Primary Tumor"] # ne garde que les tumeurs primaires
dim(se_tumor)

meta_tumor <- as.data.frame(colData(se_tumor)) # métadonnées des échantillons tumoraux
meta_tumor$patient_id <- substr(rownames(meta_tumor), 1, 12) # extraction des IDs patients

summary(table(meta_tumor$patient_id)) # permet de vérifier qu'il n'y a qu'un échantillon par patient, évite les doublons
# on peut maintenant dire que 1 colonne = 1 patient = 1 échantillon ssi "Number of factors = 1"

[1] 60660   505

Number of cases in table: 505 
Number of factors: 1 

## *Travailler sur un dataframe réduit, avec seulement les tumeurs primaires*

In [14]:
stopifnot(all(table(meta_tumor$patient_id) == 1)) # si il y a des doublons, le script s'arrête
stopifnot(all(meta_tumor$barcode %in% colnames(expr))) # vérifie que tous les échantillons tumoraux sont dans la matrice d'expression

se_patient <- se_tumor # simplifie les noms (pour pas s'y perdre)
metadata_patients <- meta_tumor

- se_patient : matrice d'expression avec seulement les échantillons "Primary Tumor"
- metadata_patient : metadata avec seulement les échantillons "Primary Tumor"

In [55]:
head(assay(se_patient))
dim(assay(se_patient))
head(metadata_patients, 3) # les 3 premiers

TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-FK-A3SE-01A-11R-A22L-07
ENSG00000000003.15 6156                         7191                        
ENSG00000000005.6     7                            3                        
ENSG00000000419.13 2540                         3815                        
ENSG00000000457.14 1452                          767                        
ENSG00000000460.17  297                          276                        
ENSG00000000938.13 1075                         1069                        
                   TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-FY-A2QD-01A-11R-A19O-07
ENSG00000000003.15 2219                         5185                        
ENSG00000000005.6     1                            0                        
ENSG00000000419.13  735                         2594                        
ENSG00000000457.14  305                          970                        
ENSG00000000460.17   28                          135                        
ENSG00000000938.13  303                           71                        
                   TCGA-EL-A3GR-01A-21R-A206-07 TCGA-EL-A3T7-01A-11R-A22L-07
ENSG00000000003.15 4466                         9702                        
ENSG00000000005.6     3                            0                        
ENSG00000000419.13 1496                         2506                        
ENSG00000000457.14  820                         1036                        
ENSG00000000460.17  208                          315                        
ENSG00000000938.13  610                         1576                        
                   TCGA-FE-A230-01A-11R-A14Y-07 TCGA-EM-A3FO-01A-11R-A21D-07
ENSG00000000003.15 2449                         6336                        
ENSG00000000005.6     0                            0                        
ENSG00000000419.13  811                         1892                        
ENSG00000000457.14  619                          725                        
ENSG00000000460.17  194                          149                        
ENSG00000000938.13  446                          315                        
                   TCGA-DJ-A3VB-01A-11R-A23N-07 TCGA-BJ-A28X-01A-11R-A22L-07 ⋯
ENSG00000000003.15 5997                         3534                         ⋯
ENSG00000000005.6     2                            1                         ⋯
ENSG00000000419.13 3074                         1769                         ⋯
ENSG00000000457.14  856                          948                         ⋯
ENSG00000000460.17  317                          332                         ⋯
ENSG00000000938.13 1829                         1886                         ⋯
                   TCGA-DJ-A13S-01A-11R-A10U-07 TCGA-EM-A3SY-01A-11R-A22U-07
ENSG00000000003.15 2448                         3693                        
ENSG00000000005.6     0                            0                        
ENSG00000000419.13 1851                         2286                        
ENSG00000000457.14  764                          459                        
ENSG00000000460.17  176                          185                        
ENSG00000000938.13  635                          332                        
                   TCGA-EM-A2P1-01A-11R-A206-07 TCGA-EL-A3T1-01A-11R-A22L-07
ENSG00000000003.15 2691                         4533                        
ENSG00000000005.6     0                            0                        
ENSG00000000419.13 2227                         2406                        
ENSG00000000457.14  759                         1120                        
ENSG00000000460.17  279                          381                        
ENSG00000000938.13 1147                         1127                        
                   TCGA-DJ-A2PS-01A-11R-A18C-07 TCGA-EL-A3TB-01A-11R-A22L-07
ENSG00000000003.15 4471                         4742                        
ENSG00000000005.6     0                            3                        
E

[1] 60660   505

barcode                      patient     
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA
                             sample           shortLetterCode
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6-01A TP             
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE-01A TP             
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA-01A TP             
                             definition          sample_submitter_id
TCGA-DJ-A2Q6-01A-11R-A18C-07 Primary solid Tumor TCGA-DJ-A2Q6-01A   
TCGA-FK-A3SE-01A-11R-A22L-07 Primary solid Tumor TCGA-FK-A3SE-01A   
TCGA-DJ-A2QA-01A-11R-A18C-07 Primary solid Tumor TCGA-DJ-A2QA-01A   
                             tumor_descriptor
TCGA-DJ-A2Q6-01A-11R-A18C-07 Primary         
TCGA-FK-A3SE-01A-11R-A22L-07 Primary         
TCGA-DJ-A2QA-01A-11R-A18C-07 Primary         
                             sample_id                           
TCGA-DJ-A2Q6-01A-11R-A18C-07 74b9c0c8-ef36-4194-ab8c-d81d61a39705
TCGA-FK-A3SE-01A-11R-A22L-07 d3cc158d-413d-49e6-8d23-3ed89fdea94a
TCGA-DJ-A2QA-01A-11R-A18C-07 2277f3f2-411e-47ab-a2b9-f80fd33540a1
                             pathology_report_uuid                submitter_id
TCGA-DJ-A2Q6-01A-11R-A18C-07 CCF825BD-7410-4ED2-B0E2-8B17BB5C5BBD TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07 3489196F-F9FD-4797-B4F2-0D66D420102A TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07 E3CC8058-B202-4433-8597-6BD196D04AAC TCGA-DJ-A2QA
                             ⋯ paper_mutDriver_CCF_gene paper_EXCLUDE_SAMPLE
TCGA-DJ-A2Q6-01A-11R-A18C-07 ⋯ BRAF:1.000               0                   
TCGA-FK-A3SE-01A-11R-A22L-07 ⋯                          0                   
TCGA-DJ-A2QA-01A-11R-A18C-07 ⋯ BRAF:1.000               0                   
                             paper_patient paper_TERT_DEPTH paper_TERT_POWER
TCGA-DJ-A2Q6-01A-11R-A18C-07 TCGA-DJ-A2Q6  81005            1               
TCGA-FK-A3SE-01A-11R-A22L-07 TCGA-FK-A3SE  44535            1               
TCGA-DJ-A2QA-01A-11R-A18C-07 TCGA-DJ-A2QA  75617            1               
                             paper_TERT paper_TERT_AF paper_TERT_AC
TCGA-DJ-A2Q6-01A-11R-A18C-07                          NA           
TCGA-FK-A3SE-01A-11R-A22L-07                          NA           
TCGA-DJ-A2QA-01A-11R-A18C-07                          NA           
                             paper_TERT_Q patient_id  
TCGA-DJ-A2Q6-01A-11R-A18C-07              TCGA-DJ-A2Q6
TCGA-FK-A3SE-01A-11R-A22L-07              TCGA-FK-A3SE
TCGA-DJ-A2QA-01A-11R-A18C-07              TCGA-DJ-A2QA

Note d'Elodie : j'ai pas fini de trier la suite du code, j'ai juste trié ce qu'il y avait avant

Note de Terry : J'ai ajouté une vérification pour être sûr que tous les échantillons tumoraux sont bien dans la matrice d'expression et aussi qu'on prends bien les "unstranded" counts qu'il faudra normaliser plus tard.
Je reprends à partir d'ici pour le plus gros.

### Choix de l'objet source

In [34]:
se_src <- if (exists("se_patient")) se_patient else se

meta0 <- as.data.frame(colData(se_src))

# si jamais barcode n'existe pas (rare), on le reconstruit depuis colnames
if (!"barcode" %in% names(meta0)) meta0$barcode <- colnames(se_src)

meta0 <- meta0 %>%
  mutate(
    barcode    = as.character(barcode),
    patient_id = substr(barcode, 1, 12)
  )

### 1) Garder seulement TP + 1 échantillon par patient

In [35]:
meta_tp <- meta0 %>%
  filter(shortLetterCode == "TP") %>%
  arrange(patient_id, barcode) %>%
  distinct(patient_id, .keep_all = TRUE)


### 2) Âge, sexe, stage, type de tumeur, puis mutation simple

In [57]:
meta_tp <- meta_tp %>%
  mutate(
    # Age
    age_years = as.numeric(age_at_diagnosis) / 365.25,
    age_yr_round = as.integer(round(age_years)),

    # Sexe
    sex_short = case_when(
      tolower(gender) %in% c("male","m")   ~ "male",
      tolower(gender) %in% c("female","f") ~ "female",
      TRUE ~ "UNKSEX"
    ),

    # Stage simplifié
    stage = as.character(ajcc_pathologic_stage),
    stage_simple = case_when(
      grepl("^Stage I\\b", stage)   ~ "Stade_I",
      grepl("^Stage II\\b", stage)  ~ "Stade_II",
      grepl("^Stage III\\b", stage) ~ "Stade_III",
      grepl("^Stage IV", stage)  ~ "Stade_IV",
      TRUE ~ NA_character_
    ),

    # Type cancer : PTC / FTC / OTHER
    diagnosis = as.character(primary_diagnosis),
    cancer_type = case_when(
      grepl("papillar",  tolower(diagnosis)) ~ "PTC",
      grepl("follicular",tolower(diagnosis)) ~ "FTC",
      TRUE ~ "OTHER"
    )
  )



### 3) Mutation + catégories

get_mutation_label <- function(df) {
  if ("paper_mutDriver_CCF_gene" %in% names(df)) {
    x <- as.character(df$paper_mutDriver_CCF_gene)   # ex "BRAF:1.000"
    mut <- sub(":.*$", "", x)
    mut <- ifelse(is.na(mut) | mut == "", "UNK", mut)
    return(mut)
  }
  rep("UNK", nrow(df))
}

meta_tp$mutation <- get_mutation_label(meta_tp)

meta_tp <- meta_tp %>%
  mutate(
    mutation_group = case_when(
      mutation == "BRAF" ~ "BRAF",
      mutation %in% c("NRAS","HRAS","KRAS") ~ "RAS",
      mutation == "EIF1AX" ~ "EIF1AX",
      mutation == "UNK" ~ "NO_CANONICAL_DRIVER",
      TRUE ~ "OTHER"
    )
  )

In [58]:
# Visualisation de ce qui a été fait avant
head(meta_tp)
# Nombre de stage, type de cancer, mutation et age
table(meta_tp$stage_simple)
table(meta_tp$cancer_type)
table(meta_tp$mutation_group)
table(meta_tp$age_yr_round)

barcode                      patient     
TCGA-4C-A93U-01A-11R-A39I-07 TCGA-4C-A93U-01A-11R-A39I-07 TCGA-4C-A93U
TCGA-BJ-A0YZ-01A-11R-A10U-07 TCGA-BJ-A0YZ-01A-11R-A10U-07 TCGA-BJ-A0YZ
TCGA-BJ-A0Z0-01A-11R-A10U-07 TCGA-BJ-A0Z0-01A-11R-A10U-07 TCGA-BJ-A0Z0
TCGA-BJ-A0Z2-01A-11R-A10U-07 TCGA-BJ-A0Z2-01A-11R-A10U-07 TCGA-BJ-A0Z2
TCGA-BJ-A0Z3-01A-11R-A13Y-07 TCGA-BJ-A0Z3-01A-11R-A13Y-07 TCGA-BJ-A0Z3
TCGA-BJ-A0Z5-01A-11R-A10U-07 TCGA-BJ-A0Z5-01A-11R-A10U-07 TCGA-BJ-A0Z5
                             sample           shortLetterCode
TCGA-4C-A93U-01A-11R-A39I-07 TCGA-4C-A93U-01A TP             
TCGA-BJ-A0YZ-01A-11R-A10U-07 TCGA-BJ-A0YZ-01A TP             
TCGA-BJ-A0Z0-01A-11R-A10U-07 TCGA-BJ-A0Z0-01A TP             
TCGA-BJ-A0Z2-01A-11R-A10U-07 TCGA-BJ-A0Z2-01A TP             
TCGA-BJ-A0Z3-01A-11R-A13Y-07 TCGA-BJ-A0Z3-01A TP             
TCGA-BJ-A0Z5-01A-11R-A10U-07 TCGA-BJ-A0Z5-01A TP             
                             definition          sample_submitter_id
TCGA-4C-A93U-01A-11R-A39I-07 Primary solid Tumor TCGA-4C-A93U-01A   
TCGA-BJ-A0YZ-01A-11R-A10U-07 Primary solid Tumor TCGA-BJ-A0YZ-01A   
TCGA-BJ-A0Z0-01A-11R-A10U-07 Primary solid Tumor TCGA-BJ-A0Z0-01A   
TCGA-BJ-A0Z2-01A-11R-A10U-07 Primary solid Tumor TCGA-BJ-A0Z2-01A   
TCGA-BJ-A0Z3-01A-11R-A13Y-07 Primary solid Tumor TCGA-BJ-A0Z3-01A   
TCGA-BJ-A0Z5-01A-11R-A10U-07 Primary solid Tumor TCGA-BJ-A0Z5-01A   
                             tumor_descriptor
TCGA-4C-A93U-01A-11R-A39I-07 Primary         
TCGA-BJ-A0YZ-01A-11R-A10U-07 Primary         
TCGA-BJ-A0Z0-01A-11R-A10U-07 Primary         
TCGA-BJ-A0Z2-01A-11R-A10U-07 Primary         
TCGA-BJ-A0Z3-01A-11R-A13Y-07 Primary         
TCGA-BJ-A0Z5-01A-11R-A10U-07 Primary         
                             sample_id                           
TCGA-4C-A93U-01A-11R-A39I-07 a5b168e1-a5e3-48d3-9c7b-006750c34d97
TCGA-BJ-A0YZ-01A-11R-A10U-07 f07142d5-f504-4760-b5cc-f58c6d93ef0a
TCGA-BJ-A0Z0-01A-11R-A10U-07 51fe3d91-4914-49fc-b69c-990c1dd09e1c
TCGA-BJ-A0Z2-01A-11R-A10U-07 b06b7493-f38a-493f-8b47-61e0fa4d111f
TCGA-BJ-A0Z3-01A-11R-A13Y-07 6bf032c1-b7d5-4b1c-bf41-c9bfeafa5e7a
TCGA-BJ-A0Z5-01A-11R-A10U-07 d17746b4-edea-45db-9ad1-5ee4973f007c
                             pathology_report_uuid                submitter_id
TCGA-4C-A93U-01A-11R-A39I-07 52F5D12B-92C9-4A98-A9EB-42BF48C0BA87 TCGA-4C-A93U
TCGA-BJ-A0YZ-01A-11R-A10U-07 B4B35251-8130-41C6-A7DB-478D8673D12C TCGA-BJ-A0YZ
TCGA-BJ-A0Z0-01A-11R-A10U-07 55206358-DB4D-4FD8-B634-A2DB15F8F7C1 TCGA-BJ-A0Z0
TCGA-BJ-A0Z2-01A-11R-A10U-07 C93FF278-A750-44B7-8A22-7AC06BDFB799 TCGA-BJ-A0Z2
TCGA-BJ-A0Z3-01A-11R-A13Y-07 EDF3F3EB-AC62-403D-9180-147C2CFD2AA5 TCGA-BJ-A0Z3
TCGA-BJ-A0Z5-01A-11R-A10U-07 F4DCEE4E-CB1C-48CB-8BE2-90A9925D8A70 TCGA-BJ-A0Z5
                             ⋯ patient_id   age_years age_yr_round sex_short
TCGA-4C-A93U-01A-11R-A39I-07 ⋯ TCGA-4C-A93U 74.87474  75           female   
TCGA-BJ-A0YZ-01A-11R-A10U-07 ⋯ TCGA-BJ-A0YZ 65.98768  66           male     
TCGA-BJ-A0Z0-01A-11R-A10U-07 ⋯ TCGA-BJ-A0Z0 55.74538  56           male     
TCGA-BJ-A0Z2-01A-11R-A10U-07 ⋯ TCGA-BJ-A0Z2 57.51677  58           male     
TCGA-BJ-A0Z3-01A-11R-A13Y-07 ⋯ TCGA-BJ-A0Z3 33.66188  34           female   
TCGA-BJ-A0Z5-01A-11R-A10U-07 ⋯ TCGA-BJ-A0Z5 58.50513  59           male     
                             stage     stage_simple
TCGA-4C-A93U-01A-11R-A39I-07 Stage IVA Stade_IV    
TCGA-BJ-A0YZ-01A-11R-A10U-07 Stage II  Stade_II    
TCGA-BJ-A0Z0-01A-11R-A10U-07 Stage II  Stade_II    
TCGA-BJ-A0Z2-01A-11R-A10U-07 Stage IVC Stade_IV    
TCGA-BJ-A0Z3-01A-11R-A13Y-07 Stage I   Stade_I     
TCGA-BJ-A0Z5-01A-11R-A10U-07 Stage III Stade_III   
                             diagnosis                      cancer_type
TCGA-4C-A93U-01A-11R-A39I-07 Papillary carcinoma, NOS       PTC        
TCGA-BJ-A0YZ-01A-11R-A10U-07 Papillary adenocarcinoma, NOS  PTC        
TCGA-BJ-A0Z0-01A-11R-A10U-07 Papillary adenocarcinoma, NOS  PTC        
TCGA-BJ-A0Z2-01A-11R-A10U-07 Follicular adenocarcinoma, NOS FTC        
TCGA-BJ-A0Z3-01


  Stade_I  Stade_II Stade_III  Stade_IV 
      264        48        99        49 


  FTC OTHER   PTC 
    2    28   475 


               BRAF              EIF1AX NO_CANONICAL_DRIVER                 RAS 
                229                   5                 220                  51 


13 15 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 
 1  2  2  1  4  3  4  5  5  5  5  2  6  9  9  5 10  9 13 10 17  9 14 12 13  8 
41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 
11 11 11  4 12 15 11  8  7 10 14 19  5 10 10 13 11  8  6  2  8  8 11  6  6  8 
67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 83 84 85 86 87 88 89 
 4  6  3  9  4  5  3  1  3  3  4  4  3  2  2  2  2  1  1  1  1  1 

### 4) Mettre les labels

In [59]:
meta_final <- meta_tp %>%
  mutate(
    sample_label = paste(
      barcode,                          # ou patient_id si vous préférez
      paste0(age_yr_round, "ANS"),
      sex_short,
      stage_simple,
      mutation_group,
      cancer_type,
      sep = "_"
    ),
    sample_label = make.unique(sample_label)
  )


## **Matrice d'expresison finale**

In [60]:
expr <- assay(se_src, "unstranded")  # recommandé pour DL (comptages bruts)
expr_final <- expr[, meta_final$barcode, drop = FALSE]
colnames(expr_final) <- meta_final$sample_label

In [61]:
head(expr_final[, 1:5])

TCGA-4C-A93U-01A-11R-A39I-07_75ANS_female_Stade_IV_NO_CANONICAL_DRIVER_PTC
ENSG00000000003.15 2967                                                                      
ENSG00000000005.6    12                                                                      
ENSG00000000419.13 1160                                                                      
ENSG00000000457.14  549                                                                      
ENSG00000000460.17  213                                                                      
ENSG00000000938.13  560                                                                      
                   TCGA-BJ-A0YZ-01A-11R-A10U-07_66ANS_male_Stade_II_NO_CANONICAL_DRIVER_PTC
ENSG00000000003.15 4676                                                                    
ENSG00000000005.6    11                                                                    
ENSG00000000419.13 1184                                                                    
ENSG00000000457.14  878                                                                    
ENSG00000000460.17  124                                                                    
ENSG00000000938.13  310                                                                    
                   TCGA-BJ-A0Z0-01A-11R-A10U-07_56ANS_male_Stade_II_NO_CANONICAL_DRIVER_PTC
ENSG00000000003.15 3163                                                                    
ENSG00000000005.6     2                                                                    
ENSG00000000419.13 1307                                                                    
ENSG00000000457.14 1271                                                                    
ENSG00000000460.17  100                                                                    
ENSG00000000938.13  136                                                                    
                   TCGA-BJ-A0Z2-01A-11R-A10U-07_58ANS_male_Stade_IV_RAS_FTC
ENSG00000000003.15 2786                                                    
ENSG00000000005.6     1                                                    
ENSG00000000419.13 1330                                                    
ENSG00000000457.14  667                                                    
ENSG00000000460.17  130                                                    
ENSG00000000938.13  137                                                    
                   TCGA-BJ-A0Z3-01A-11R-A13Y-07_34ANS_female_Stade_I_BRAF_PTC
ENSG00000000003.15 5626                                                      
ENSG00000000005.6     0                                                      
ENSG00000000419.13 2195                                                      
ENSG00000000457.14 1309                                                      
ENSG00000000460.17  351                                                      
ENSG00000000938.13 1214

In [ ]:
# 6) Mapping ENSG -> gene_name (provisoire)
rd <- as.data.frame(rowData(se_src))

# très souvent dispo ici:
gene_name <- rd$havana_gene
gene_name[is.na(gene_name) | gene_name == ""] <- rownames(expr_final)

# je conseille de garder ENSG + ajouter gene_name à côté,
# mais si ton prof veut vraiment les rownames en gene symbols :
rownames(expr_final) <- make.unique(gene_name)

# Check final
dim(expr_final)
head(colnames(expr_final), 3)
head(rownames(expr_final), 3)

Warning message in gene_name[is.na(gene_name) | gene_name == ""] <- rownames(expr_final):
"le nombre d'objets à remplacer n'est pas multiple de la taille du remplacement"


[1] 60660   505

[1] "TCGA-4C-A93U-01A-11R-A39I-07_75ANS_female_Stade_IV_NO_CANONICAL_DRIVER_PTC"
[2] "TCGA-BJ-A0YZ-01A-11R-A10U-07_66ANS_male_Stade_II_NO_CANONICAL_DRIVER_PTC"  
[3] "TCGA-BJ-A0Z0-01A-11R-A10U-07_56ANS_male_Stade_II_NO_CANONICAL_DRIVER_PTC"

[1] "OTTHUMG00000022002.2" "OTTHUMG00000022001.2" "OTTHUMG00000032742.2"

In [64]:
head(expr_final)
dim(expr_final)

TCGA-4C-A93U-01A-11R-A39I-07_75ANS_female_Stade_IV_NO_CANONICAL_DRIVER_PTC
OTTHUMG00000022002.2 2967                                                                      
OTTHUMG00000022001.2   12                                                                      
OTTHUMG00000032742.2 1160                                                                      
OTTHUMG00000035941.6  549                                                                      
OTTHUMG00000035821.9  213                                                                      
OTTHUMG00000003516.3  560                                                                      
                     TCGA-BJ-A0YZ-01A-11R-A10U-07_66ANS_male_Stade_II_NO_CANONICAL_DRIVER_PTC
OTTHUMG00000022002.2 4676                                                                    
OTTHUMG00000022001.2   11                                                                    
OTTHUMG00000032742.2 1184                                                                    
OTTHUMG00000035941.6  878                                                                    
OTTHUMG00000035821.9  124                                                                    
OTTHUMG00000003516.3  310                                                                    
                     TCGA-BJ-A0Z0-01A-11R-A10U-07_56ANS_male_Stade_II_NO_CANONICAL_DRIVER_PTC
OTTHUMG00000022002.2 3163                                                                    
OTTHUMG00000022001.2    2                                                                    
OTTHUMG00000032742.2 1307                                                                    
OTTHUMG00000035941.6 1271                                                                    
OTTHUMG00000035821.9  100                                                                    
OTTHUMG00000003516.3  136                                                                    
                     TCGA-BJ-A0Z2-01A-11R-A10U-07_58ANS_male_Stade_IV_RAS_FTC
OTTHUMG00000022002.2 2786                                                    
OTTHUMG00000022001.2    1                                                    
OTTHUMG00000032742.2 1330                                                    
OTTHUMG00000035941.6  667                                                    
OTTHUMG00000035821.9  130                                                    
OTTHUMG00000003516.3  137                                                    
                     TCGA-BJ-A0Z3-01A-11R-A13Y-07_34ANS_female_Stade_I_BRAF_PTC
OTTHUMG00000022002.2 5626                                                      
OTTHUMG00000022001.2    0                                                      
OTTHUMG00000032742.2 2195                                                      
OTTHUMG00000035941.6 1309                                                      
OTTHUMG00000035821.9  351                                                      
OTTHUMG00000003516.3 1214                                                      
                     TCGA-BJ-A0Z5-01A-11R-A10U-07_59ANS_male_Stade_III_NO_CANONICAL_DRIVER_PTC
OTTHUMG00000022002.2 2424                                                                     
OTTHUMG00000022001.2    4                                                                     
OTTHUMG00000032742.2 1374                                                                     
OTTHUMG00000035941.6  757                                                                     
OTTHUMG00000035821.9  255                                                                     
OTTHUMG00000003516.3  709                                                                     
                     TCGA-BJ-A0Z9-01A-11R-A10U-07_58ANS_female_Stade_III_BRAF_PTC
OTTHUMG00000022002.2 1732                                                        
OTTHUMG00000022001.2    0                                                        
OTTHUMG000000327

[1] 60660   505